### PR TITLE
added dress search API

### PR DESCRIPTION
**Pull request by Max-Y-Huang**

### CODE REVIEW

File: php/dresssearch.php [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        echo "{\"status\":400, \"result\":\"The request type must bt GET\"}";
+        die();
+    }
+    elseif (empty($_GET) || !isset($_GET['token']) || !isset($_GET['max-search']))
+    {
+        http_response_code(400);
+        echo "{\"status\":400, \"result\":\"Missing token or required fields\"}";
+        die();
+    }
+    elseif (!isset($_GET['name']) && !isset($_GET['colour']) && !isset($_GET['brand']))
```
Comment (cheeseburgur101) [inline php/dresssearch.php]: Wouldn't it be better to let people search using any combination of these? I.e. a user wanting to search for a particular brand, regardless of color.

Comment (Max-Y-Huang) [reply php/dresssearch.php]: That's how it works though

Comment (Max-Y-Huang) [reply php/dresssearch.php]: If none of the fields are filled, an error is returned.

Comment (cheeseburgur101) [reply php/dresssearch.php]: ah I see, nevermind

**Pull request merged**